meson: Versioning for git builds

Rafostar opened the pull request:
When building from untagged or dirty git repo, extend version information with git commit timestamp and SHA. If dirty, use current timestamp and also append "-dirty" suffix.

This helps to determine which exact commit is used and ensures that enhancers cache is always rebuild during development for easier testing.

Rafostar committed:
meson: Versioning for git builds

When building from untagged or dirty git repo, extend version
information with git commit timestamp and SHA. If dirty, use
current timestamp and also append "-dirty" suffix.

This helps to determine which exact commit is used and ensures that
enhancers cache is always rebuild during development for easier testing.

Rafostar commented:
@sp1ritCS

This seems more "sane" than #24.

To sum-up: version is left as it was instead changing to a date. When during build meson detects build from cloned git repo, but HEAD is not a tagged commit (or modified), then `+git.%ct~%h` is added in each plugin version info file (which is the same syntax as used in OBS repo).

The aim is for both to be able too see in Clapper preferences that git/unstable build of enhancers is used AND changed version ensures that Clapper will reload given enhancer which makes testing during development much easier.

sp1ritCS commented:
@Rafostar ack.

Keep in mind that OBS packages have no access to git metadata, so I guess they'll continue to show as 0.9.0.

Rafostar commented:
> @Rafostar ack.
> 
> Keep in mind that OBS packages have no access to git metadata, so I guess they'll continue to show as 0.9.0.

Yes, it will stay simply 0.9.0 as fallback. But I do wonder if I can improve that for OBS separately...
I was thinking of adding a meson option to allow to define custom version, then in OBS something like this should work (I hope):

```sh
%meson -Denhancersver=%{version}
```

since AFAIU `_service` updates `%{version}` part of specfile before run.

Rafostar committed:
meson: Allow to specify custom version

Can by used by CI to override default version string.
Useful when "git" is not available at compile time.
